client/src/components/App.jsx

```diff
@@ -64,25 +64,53 @@ const App = () => {
 
   // API calls
   const getProducts = async () => {
-    const fetchedProducts = await axios.get('/products');
-    return fetchedProducts.data;
+    const cachedProducts = localStorage.getItem('products');
```

Comment:
const cachedProducts = await localStorage.getItem('products') ?

Reply:
Not sure how async the getItem function is, but assuming it is async

Reply:
No, it seems to be synchronous. Works without specifying that it's async, and stackOverflow seems to think it's synchronous

Reply:
errata: there's something else call `asyncLocalStorage()` which is apparently faster: https://dev.to/createnextapp/why-asynclocalstorage-instead-of-localstorage-1ha7
Will look into it